### PR TITLE
Refactor for issue #4

### PR DESCRIPTION
**Pull request by joefearnley**
Hi @peter279k,

I've refactored this library quite a bit to accommodate for the TODO issue #4. I extracted all of the services out into separate classes that implement a Service interface. Then using the factory pattern, I created a ServiceFactory class that does the creating of those service class objects. 

The interface has changed a bit. To create a service objects, you would use the ServiceFactory class like so:

```
$config = [
    'api-key' => 'key-1234567890',
    'other-setting' => '1234567890'
];

$exampleService = (new ServiceFactory)->create('name-of-service');
$mailgunService->setConfig($config);
$response = $mailgunService->sendReq();
``` 

See the tests for examples. 

Something else we could do, if you want to go this route, is make the create method static so it would something like this:

`$exampleService = ServiceFactory::create('name-of-service');`

I also created a tests directory and a test for each class. 

If you want to accept this pull request, let me know and I will update the README to work with the this interface.

Thanks!

### CODE REVIEW

File: src/serviceBundle/ServiceFactory.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use peter\components\serviceBundle\Services\Bitly;
+use peter\components\serviceBundle\Services\McAfee;
+use peter\components\serviceBundle\Services\Google;
+use peter\components\serviceBundle\Services\SendGrid;
+
+class ServiceFactory
+{
+    public function create($service) 
+    {
+        switch ($service) {
```
Comment (peter279k) [inline src/serviceBundle/ServiceFactory.php]: What do you think using the following sample code to replace the switch....case?

```php
$service = 'MailGun';
$object = new $service();
```

File: src/serviceBundle/Services/Bitly.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        $this->config = $config;
+    }
+
+    public function sendReq()
+    {
+        $url = 'http://api.bit.ly/v3/shorten?login='.$this->config['login'].'&apiKey='.$this->config['apiKey'].'&uri='.urlencode($this->config['longUrl']);
+        $client = new \GuzzleHttp\Client();
+        return $client->get($url)->json();
+    }
+}
```
Comment (peter279k) [inline src/serviceBundle/Services/Bitly.php]: Missing the EOF.

**Review by peter279k — CHANGES_REQUESTED** [review]

**Comment by peter279k** [comment]
Hi @joefearnley, thank you for your PR.
There're some request changes that you need to consider.
Don't be worried about the Travis-CI error, but the coding style(StyleCI) you have to fix this.
And the Travis-CI error I will solve.

Thanks.

File: src/serviceBundle/ServiceFactory.php (hunk 1 above)
Comment (joefearnley) [reply src/serviceBundle/ServiceFactory.php]: Yes. Let me look at the options for replacing the switch statement.

**Comment by joefearnley** [comment]
OK. I will get these changes made.

Another thing I forgot to mentions in the first comment - I have not figured out how to hide API keys in tests. For the Mailgun and Sendgrid tests I just mocked the API so the tests will pass. Might be a good idea to utilize environment variables for this? Let me know what you think.

**Comment by peter279k** [comment]
Hi @joefearnley, thank you for your reply.
The API testing are two approaches.
Using the Mock to pass the HTTP testing and do the real testing.

The mock approach is simple and good, but it cannot guarantee that it will get the real result.
To keep the test simple, I prefer the mock approach from now on.
The services are not little and maybe the bundle will add other service in the future.

**Comment by joefearnley** [comment]
Alright, @peter279k I refactored the ServiceFactory class. It turns out class_exists() needs the namespace. That or I am not doing something right with namespaces. 

Also, I changed the Service interface to an abstract class. That way I could remove the setConfig() method from every service class. 

Oh, and I added a newline to every file. 

Hope this helps!

**Comment by peter279k** [comment]
Hi @joefearnley, thank you for your latest commit.
It looks good for me.
One more thing the StyleCI is still failed during checking coding style.
Would you fix them?
Once fixing coding style, I will merge the PR.

Thanks.

**Comment by joefearnley** [comment]
Hi @peter279k. I agree, mocking the API calls certainly do not guarantee that the results are correct. I'll mock the calls from now on until you specify otherwise.

I _think_ I have made all of the changes style ci requested. If not, please let me know.

**Comment by peter279k** [comment]
Great! @joefearnley, thank you for this PR.
If you want to improve the testing, please feel free to create another PR.

Thanks.

**Pull request merged**